src/release-notes-generator.js

```diff
@@ -1,35 +1,80 @@
-const getStream = require("get-stream");
-const intoStream = require("into-stream");
-const { sync: parser } = require("conventional-commits-parser");
-const writer = require("conventional-changelog-writer");
-const filter = require("conventional-commits-filter");
-const loadChangelogConfig = require("@semantic-release/release-notes-generator/lib/load-changelog-config");
-const { getCommit } = require("./utils");
-
-const generateNotes = async (strategy, pluginConfig, context) => {
+import getStream from "get-stream";
+import intoStream from "into-stream";
+import { sync as parser } from "conventional-commits-parser";
+import filter from "conventional-commits-filter";
+import writer from "conventional-changelog-writer";
+
+import loadChangelogConfig from "./lib/load-changelog-config.js";
+import { getCommit } from "./utils.js";
+
+import debugFactory from "debug";
+const debug = debugFactory(
+  "semantic-release:pr-analyzer:release-notes-generator"
+);
+
+export async function generateNotes(strategy, pluginConfig, context) {
   const commit = await getCommit(strategy, context.commits);
 
   const { parserOpts, writerOpts } = await loadChangelogConfig(
     pluginConfig,
     context
   );
 
-  const parsedCommits = filter([
-    {
-      ...commit,
-      ...parser(commit.message, {
-        ...parserOpts,
-      }),
-    },
-  ]);
-
   const changelogContext = {
     version: context.nextRelease.version,
   };
 
+  const parsedCommits = filter(
+    [commit]
+      .filter(({ message, hash }) => {
+        if (!message.trim()) {
+          debug("Skip commit %s with empty message", hash);
+          return false;
+        }
+
+        return true;
+      })
+      .map((rawCommit) => {
+        const parsed = parser(rawCommit.message, { ...parserOpts });
+        // Remove undefined date fields from parsed to avoid overwriting valid dates from rawCommit
+        const { committerDate: _, authorDate: __, ...parsedWithoutDates } = parsed;
+        const result = {
+          ...rawCommit,
+          ...parsedWithoutDates,
+        };
+        debug("Parsed commit before writer:", JSON.stringify({
+          hash: result.hash,
+          committerDate: result.committerDate,
+          authorDate: result.authorDate,
+          committerDateType: typeof result.committerDate,
+          authorDateType: typeof result.authorDate,
+        }));
+        return result;
```

Comment:
add here if we need to removed the udnef's
```suggestion
    if (commit.committerDate === undefined) {
      delete commit.committerDate;
    }
    if (commit.authorDate === undefined) {
      delete commit.authorDate;
    }
        return result;
```